pkgs/top-level/python-packages.nix

```diff
@@ -19123,6 +19123,29 @@ let
     };
   };
 
+  cmdtest = buildPythonPackage rec {
+    name = "cmdtest-${version}";
+    version = "0.15";
+    disabled = isPy3k || isPyPy;
+
+    buildInputs = with self; [ cliapp ttystatus ];
+    propagatedBuildInputs = with self; [ cliapp ttystatus markdown ];
+
+    # TODO: cmdtest tests must be run before the buildPhase
```

Comment:
Testing requires another package CoverageTestRunner[1] which is not yet in nixpkgs, I have a patch ready to add it based on top of this patch, however that won't be enough, the cmdtest tests need to be run preBuild (so not with doCheck presumably?) and upstream must be patched else the test will fail when it tries to remove a .coverage file, I plan to send a patch to fix that to upstream.

[1]: http://liw.fi/coverage-test-runner/